### PR TITLE
cmd/puppeth: new version as presented at devcon3

### PR DESCRIPTION
**Pull request by karalabe**
Ideal to have before merging:

 - [x] Containerize geth-alltools to have a reliable `faucet` source without needing an external repo (puppeth/go-ethereum).
 - [ ] Check and fix and issues with web deployments without a reverse-proxy.

### CODE REVIEW

File: cmd/puppeth/module_explorer.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	"math/rand"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// explorerDockerfile is the Dockerfile required to run a block explorer.
+var explorerDockerfile = `
```
Comment (gballet) [inline cmd/puppeth/module_explorer.go]: I see that you have a lot of these files embedded into go files. How about using something like Ego ? https://github.com/benbjohnson/ego

File: cmd/puppeth/module_node.go [hunk 2]
```diff
@@ -164,14 +170,43 @@ type nodeInfos struct {
 	gasPrice   float64
 }
 
-// String implements the stringer interface.
-func (info *nodeInfos) String() string {
-	discv5 := ""
+// Report converts the typed struct into a plain string->string map, cotnaining
```
Comment (gballet) [inline cmd/puppeth/module_node.go]: typo: "cotnaining"

File: cmd/puppeth/wizard_faucet.go [hunk 3]
```diff
@@ -46,9 +47,10 @@ func (w *wizard) deployFaucet() {
 			minutes: 1440,
 			tiers:   3,
 		}
+		existed = false
```
Comment (gballet) [inline cmd/puppeth/wizard_faucet.go]: so basically

```
existed := err == nil
```

File: cmd/puppeth/module_node.go (hunk 2 above)
Comment (karalabe) [reply cmd/puppeth/module_node.go]: Good catch, fixed all occurrences.

File: cmd/puppeth/wizard_faucet.go (hunk 3 above)
Comment (karalabe) [reply cmd/puppeth/wizard_faucet.go]: Yup, much cleaner, fixed all occurrences.

File: cmd/puppeth/module_explorer.go (hunk 1 above)
Comment (karalabe) [reply cmd/puppeth/module_explorer.go]: I did consider using external file templates and perhaps embedding them with `go generate` and `go-bindata`. But truth be told these are fairly small files, so it's less of a hassle to just keep them embedded directly and have a single place to edit and debug them.

**Comment by karalabe** [comment]
@gballet Thanks for the review! PTAL :)

**Pull request merged**